streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStoreTest.java

```diff
@@ -402,6 +411,46 @@ public void shouldRespectBulkLoadOptionsDuringInit() {
         }
     }
 
+    @Test
+    public void shouldLogAndMeasureExpiredRecords() {
+        LogCaptureAppender.setClassLoggerToDebug(RocksDBSegmentedBytesStore.class);
+        final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
```

Comment:
Hmm.. could we just add a `create()` function without register, if that is the only reason we need to explicitly `unregister` it later?

Reply:
no, we need to both create and register it, or we won't capture any logs.

Reply:
n.b., re reason we have to do all the stuff (set to debug, create, register, unregister) inside the static methods is that it's normally forbidden to mess directly with log4j classes (because we should normally use slf4j as our interface).

But these tests are a special case, and in that context, it's safe to treat it as a given that the logger is actually log4j. Accordingly, I added an exception permitting us to use log4j classes inside the LogCaptureAppender.

Reply:
also, the only reason we "need" to unregister it is to avoid polluting the rest of the tests. Otherwise, the appender would continue to capture logs until the JVM exits.

Reply:
Ack, thanks!